pkg/resourcemanager/controller/tokeninvalidator/reconciler.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tokeninvalidator
+
+import (
+	"context"
+	"fmt"
+
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type reconciler struct {
+	targetClient                        client.Client
+	invalidateAllDefaultServiceAccounts bool
+}
+
+var _ reconcile.Reconciler = &reconciler{}
+
+// NewReconciler returns a new reconciler.
+func NewReconciler(targetClient client.Client, invalidateAllDefaultServiceAccounts bool) reconcile.Reconciler {
+	return &reconciler{
+		targetClient:                        targetClient,
+		invalidateAllDefaultServiceAccounts: invalidateAllDefaultServiceAccounts,
+	}
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := log.FromContext(ctx)
```

Comment:
`Variable 'log' collides with imported package name`
```suggestion
	log := logf.FromContext(ctx)
```